infrastructure/src/test/java/org/corfudb/infrastructure/health/HealthMonitorTest.java

```diff
@@ -395,5 +412,86 @@ void testGenerateHealthReport() {
         HealthMonitor.shutdown();
     }
 
+    @Test
+    @SuppressWarnings("checkstyle:magicnumber")
+    void testDeadlockDetection() {
```

Comment:
It would be great if this test can be made more deterministic. Here are a few suggestions:

- A pair of latches or similar synchronization primitives can be used instead of the Thread.sleep on L447 to deterministically create the interleaving of locking events for `block1` and `block2`.
- Instead of sleeping on L470, it looks like ReentrantLock::hasQueuedThreads can be polled to determine if any threads are waiting to acquire these locks.
- Instead of sleeping on L492, there could be indicator variables that are set in the catch blocks of `block1` and `block2`. These can be polled.
- Consider having an explicit way to trigger checkLiveness for testing to avoid races where the thread might be delayed.

Reply:
@zfrenette 
Thanks for suggestions, done